napari/layers/_labels_layer/model.py

```diff
@@ -199,7 +380,143 @@ def method(self):
     def method(self, method):
         self._node.method = method
 
-    def get_value(self, position, indices):
+    def fill(self, indices, coord, old_label, new_label):
+        """Replace an existing label with a new label, either just at the
+        connected component if the `contiguous` flag is `True` or everywhere
+        if it is `False`, working either just in the current slice if
+        the `n_dimensional` flag is `False` or on the entire data if it is
+        `True`.
+
+        Parameters
+        ----------
+        indices : sequence of int or slice
+            Indices that make up the slice.
+        coord : sequence of int
+            Position of mouse cursor in image coordinates.
+        old_label : int
+            Value of the label image at the coord to be replaced.
+        new_label : int
+            Value of the new label to be filled in.
+        """
+        int_coord = copy(coord)
+        int_coord[0] = int(round(coord[0]))
+        int_coord[1] = int(round(coord[1]))
+
+        if self.n_dimensional or self._raw_image.ndim == 2:
+            # work with entire image
+            labels = self._raw_image
+            displayed = self._image
+            slice_coord = tuple(int_coord)
+        else:
+            # work with just the sliced image
+            labels, slice_indices = self._slice_image(indices,
+                                                      image=self._raw_image)
+            slice_coord = tuple(int_coord[:2])
+            displayed = self._image[slice_indices]
+
+        matches = labels == old_label
+        if self.contiguous:
+            # if not contiguous replace only selected connected component
+            labeled_matches, num_features = ndi.label(matches)
+            if num_features != 1:
+                match_label = labeled_matches[slice_coord]
+                matches = np.logical_and(matches,
+                                         labeled_matches == match_label)
+
+        # Replace target pixels with new_label
+        labels[matches] = new_label
+        if new_label == 0:
+            displayed[matches] = 0
+        else:
+            displayed[matches] = self.raw_to_displayed(new_label)
+
+        if not (self.n_dimensional or self._raw_image.ndim == 2):
+            # if working with just the slice, update the rest of the raw image
+            self._raw_image[slice_indices] = labels
+            self._image[slice_coord] = displayed
+
+        self.refresh()
+
+    def _to_pix(self, pos, axis):
+        """Round float from cursor position to a valid pixel
+
+        Parameters
+        ----------
+        pos : float
+            Float that is to be mapped.
+        axis : 0 | 1
+            Axis that pos corresponds to.
+        Parameters
+        ----------
+        pix : int
+            Rounded pixel value
+        """
+
+        pix = int(np.clip(round(pos), 0, self.shape[axis]))
```

Comment:
should this be `self.shape[axis] - 1`? Or you might try to write past the end of the array?

Reply:
fixed